File: C_files/Lecture_1/cash/cash.c
#include <stdio.h>

int main()
{
    int change;

    // Get correct input from user
    do
    {
        printf("Change owed: ");
        scanf("%d", &change);
    }
    while (change <= 0);

    // Calculate number of quarters
    int quarters = 0;
    while (change >= 25)
    {
        change = change - 25;
        quarters++;
    }

    // Calculate number of dimes
    int dimes = 0;
    while (change >= 10)
    {
        change = change - 10;
        dimes++;
    }

    // Calculate number of nickels
    int nickels = 0;
    while (change >= 5)
    {
        change = change - 5;
        nickels++;
    }

    // Calculate number of pennies
    int pennies = 0;
    while (change >= 1)
    {
        change = change - 1;
        pennies++;
    }

    // Calculate total number of coins and print
    int count = quarters + dimes + nickels + pennies;
    printf("%d\n", count);

    return 0;
}

File: Python_files/Lecture_6/sentimental-cash/cash.py
# TODO
from cs50 import get_float

# Get user input
while True:
    cash = get_float('Change owed: ')
    if cash > 0:
        break

# Convert to cents
cents = cash * 100

# Calculate number of quarters
quarters = int(cents / 25)
cents = cents - quarters * 25

# Calculate number of dimes
dimes = int(cents / 10)
cents = cents - dimes * 10

# Calculate number of nickels
nickels = int(cents / 5)
cents = cents - nickels * 5

# Calculate number of pennies
pennies = int(cents / 1)

# Calculate number of coins to be returned
coins = quarters + dimes + nickels + pennies

# Print the results
print(f'{coins}')